src/lava/lib/optimization/apps/scheduler/problems.py

```diff
@@ -0,0 +1,310 @@
+        self._num_tasks = num_tasks
+        self._task_ids = range(num_tasks)
+        self._task_attrs = None
+        self._sat_cutoff = sat_cutoff
+        self.graph = None
+        self.adjacency = None
+        self._random_seed = seed
+
+    @property
+    def num_agents(self):
+        return self._num_agents
+
+    @num_agents.setter
+    def num_agents(self, val: int):
+        self._num_agents = val
+
+    @property
+    def agent_ids(self):
+        return self._agent_ids
+
+    @property
+    def agent_attrs(self):
+        return self._agent_attrs
+
+    @agent_attrs.setter
+    def agent_attrs(self, attr_vec):
+        self._agent_attrs = attr_vec
+
+    @property
+    def num_tasks(self):
+        return self._num_tasks
+
+    @num_tasks.setter
+    def num_tasks(self, val: int):
+        self._num_tasks = val
+
+    @property
+    def task_ids(self):
+        return self._task_ids
+
+    @property
+    def task_attrs(self):
+        return self._task_attrs
+
+    @task_attrs.setter
+    def task_attrs(self, attr_vec):
+        self._task_attrs = attr_vec
+
+    @property
+    def sat_cutoff(self):
+        return self._sat_cutoff
+
+    @sat_cutoff.setter
+    def sat_cutoff(self, val: float):
+        self._sat_cutoff = val
+
+    @property
+    def random_seed(self):
+        return self._random_seed
+
+    @random_seed.setter
+    def random_seed(self, val: int):
+        self._random_seed = val
+
+    def is_node_valid(self, *args):
+        return True
+
+    def is_edge_conflicting(self, node1, node2):
+        nodes = self.graph.nodes
+        is_same_agent = (nodes[node1]["agent_id"] == nodes[node2]["agent_id"])
+        is_same_task = (nodes[node1]["task_id"] == nodes[node2]["task_id"])
+        return True if is_same_agent or is_same_task else False
+
+    def generate(self, seed=None):
+        """ Generate a new scheduler problem. """
+        if seed != self.random_seed:  # set seed only if it's different
+            self.random_seed = seed
+            np.random.seed(seed)
+        self.graph = ntx.Graph()
+        self.generate_valid_nodes()
+        self.generate_edges_from_constraints()
+        self.rescale_adjacency()
+
+    def generate_valid_nodes(self):
+        node_id = 0
+        if self.agent_attrs is None:
+            self.agent_attrs = np.reshape(self.agent_ids,
+                                          (len(self.agent_ids), 1))
+        agent_id_attr_map = dict(zip(self.agent_ids, self.agent_attrs))
+        if self.task_attrs is None:
+            self.task_attrs = (
+                np.tile(np.reshape(self.task_ids,
+                                   (len(self.task_ids), 1)), (1, 2)))
+        task_id_attr_map = dict(zip(self.task_ids, self.task_attrs))
+        for aid, a_attr in agent_id_attr_map.items():
+            for tid, t_attr in task_id_attr_map.items():
+                if self.is_node_valid(aid, tid):
+                    self.graph.add_node(node_id,
```

Comment:
Is there any way that the graph is not empty when calling this method, so that it must be re-initiated before adding nodes?

Reply:
In addition, if you need to add a lot of edges (which will be the case here), it might be more efficient to buffer all edges that you want to add and then call add_edges_from to add all of them simultaneously.

Reply:
This method is called right after creating a fresh instance of the graph in `generate` method. So the graph is always empty in the way this method is used. But yes, there is a chance that someone uses the method outside of its intended purpose.

I made this and its sister methods notionally private by adding an underscore in front of their names.